tests/framework/client/node.go

```diff
@@ -0,0 +1,134 @@
+}
+
+func (c *nodes) Clean() {
+	for _, node := range createdNodes {
+		err := c.Delete(node.ctx, node.name, metav1.DeleteOptions{})
+		if err != nil && !errors.IsNotFound(err) {
+			panic(err)
+		}
+	}
+	for name, node := range nodeSpecMap {
+		currentNode, err := c.node.Get(context.Background(), name, metav1.GetOptions{})
+		util.PanicOnError(err)
+		currentNode.Spec = node.spec
+		currentNode.Labels = node.labels
+		_, err = c.node.Update(context.Background(), currentNode, metav1.UpdateOptions{})
+		util.PanicOnError(err)
+	}
+}
+
+func (c *nodes) Get(ctx context.Context, name string, options metav1.GetOptions) (result *v1.Node, err error) {
+	return c.node.Get(ctx, name, options)
+}
+
+func (c *nodes) List(ctx context.Context, opts metav1.ListOptions) (result *v1.NodeList, err error) {
+	return c.node.List(ctx, opts)
+}
+
+func (c *nodes) Watch(ctx context.Context, opts metav1.ListOptions) (watch.Interface, error) {
+	return c.node.Watch(ctx, opts)
+}
+
+func (c *nodes) Create(ctx context.Context, node *v1.Node, opts metav1.CreateOptions) (result *v1.Node, err error) {
+	created, err := c.node.Create(ctx, node, opts)
+	if err == nil && opts.DryRun == nil {
+		createdNodes[node.Name] = nodeInfo{created.Name, ctx}
+	}
+	return created, err
+}
+
+func (c *nodes) Update(ctx context.Context, node *v1.Node, opts metav1.UpdateOptions) (result *v1.Node, err error) {
+	if opts.DryRun == nil {
+		c.saveNodeSpecs(ctx, node.Name)
+	}
+	return c.node.Update(ctx, node, opts)
+}
+
+func (c *nodes) UpdateStatus(ctx context.Context, node *v1.Node, opts metav1.UpdateOptions) (result *v1.Node, err error) {
+	return c.node.UpdateStatus(ctx, node, opts)
+}
+
+func (c *nodes) Delete(ctx context.Context, name string, opts metav1.DeleteOptions) error {
+	return c.node.Delete(ctx, name, opts)
+}
+
+func (c *nodes) DeleteCollection(ctx context.Context, opts metav1.DeleteOptions, listOpts metav1.ListOptions) error {
+	return c.node.DeleteCollection(ctx, opts, listOpts)
+}
+
+func (c *nodes) Patch(ctx context.Context, name string, pt types.PatchType, data []byte, opts metav1.PatchOptions, subresources ...string) (result *v1.Node, err error) {
+	if opts.DryRun == nil {
+		c.saveNodeSpecs(ctx, name)
+	}
+	return c.node.Patch(ctx, name, pt, data, opts, subresources...)
+}
+
+func (c *nodes) Apply(ctx context.Context, node *corev1.NodeApplyConfiguration, opts metav1.ApplyOptions) (result *v1.Node, err error) {
+	return c.node.Apply(ctx, node, opts)
+}
+
+func (c *nodes) ApplyStatus(ctx context.Context, node *corev1.NodeApplyConfiguration, opts metav1.ApplyOptions) (result *v1.Node, err error) {
+	return c.node.ApplyStatus(ctx, node, opts)
+}
+
+func (c *nodes) PatchStatus(ctx context.Context, nodeName string, data []byte) (*v1.Node, error) {
+	return c.node.PatchStatus(ctx, nodeName, data)
+}
+
+func (c *nodes) saveNodeSpecs(ctx context.Context, name string) {
+	//check that changes are not referred to a previous created node
+	if _, exist := createdNodes[name]; !exist {
```

Comment:
I would probably negate this. If we are changing created node for a test we don't need to keep the previous state.

Reply:
Probably the comment is misleading. Here we check that the node is gonna to be modified is not a previous created one:
if we want to patch a node `x` and `x` was manually created during the test I'm not going to save the state of `x` because I know that `x` will be deleted at the end of the test.

Reply:
Exactly I was just suggesting doing `if _, exist := createdNodes[name]; exist { //node will be deleted return}`. Just suggestion.

Reply:
Oh I got it. Sorry, my bad. Agree

Reply:
Done